pmm-app/src/pmm-check/CheckPanel.tsx

```diff
@@ -1,148 +1,90 @@
-    }, 10000);
-  }
-
-  async fetchAlerts(): Promise<void> {
-    this.setState({ isLoading: true });
-
-    try {
-      const dataSource = await CheckService.getActiveAlerts();
-
-      this.setState({ dataSource });
-    } catch (err) {
       console.error(err);
     } finally {
-      this.setState({ isLoading: false });
+      setGetSettingsPending(false);
     }
-  }
-
-  render() {
-    const {
-      options: { title },
-    } = this.props;
-    const {
-      dataSource, isSttEnabled, isLoading, hasNoAccess, isRunChecksRequestPending,
-    } = this.state;
-
-    return (
-      <div className={styles.panel} data-qa="db-check-panel">
-        {isLoading ? (
-          <div className={styles.spinner}>
-            <Spinner />
-          </div>
-        ) : (
-          <>
-            <div className={styles.header}>
-              <div className={styles.title} data-qa="db-check-panel-title">
-                {title}
-              </div>
-              <ButtonWithSpinner
-                onClick={this.handleRunChecksClick}
-                isLoading={isRunChecksRequestPending}
-                disabled={hasNoAccess}
-                className={styles.runChecksButton}
-              >
-                {Messages.runDbChecks}
-              </ButtonWithSpinner>
-            </div>
-            <AlertsReloadContext.Provider value={{ fetchAlerts: this.fetchAlerts }}>
-              <Table
-                data={dataSource}
-                columns={COLUMNS}
-                isSttEnabled={isSttEnabled}
-                hasNoAccess={hasNoAccess}
+  };
+
+  useEffect(() => {
+    getSettings();
+  }, []);
+
+  const tabs = useMemo<TabEntry[]>(() => [
+    {
+      label: Messages.failedChecksTitle,
+      key: TabKeys.failedChecks,
+      component: <FailedChecksTab hasNoAccess={hasNoAccess} isSttEnabled={isSttEnabled} />,
+    },
+    {
+      label: Messages.allChecksTitle,
+      key: TabKeys.allChecks,
+      component: <AllChecksTab />,
+    },
+  ],
+  [hasNoAccess, isSttEnabled]);
+
+  return (
+    <div className={styles.panel} data-qa="db-check-panel">
+      {getSettingsPending ? (
+        <div className={styles.spinner}>
+          <Spinner />
+        </div>
+      ) : (
+        <>
+          <TabsBar className={styles.tabBar} data-qa="db-check-tabs-bar">
+            {tabs.map((tab) => (
+              <Tab
+                key={tab.key}
+                label={tab.label}
+                active={tab.key === activeTab}
+                onChangeTab={() => setActiveTab(tab.key)}
               />
-            </AlertsReloadContext.Provider>
-          </>
-        )}
-      </div>
-    );
-  }
-}
-
-export const CheckPanelRouter: FC<CheckPanelProps> = (props) => (
+            ))}
+          </TabsBar>
+          <TabContent className={styles.tabContent} data-qa="db-check-tab-content">
+            {tabs.map((tab) => tab.key === activeTab && tab.component)}
```

Comment:
nit: it's succinct, but hard to reason about. We'd better put a return statement.

Reply:
It's a copy-paste, I propose to do it in one pass everywhere it's needed